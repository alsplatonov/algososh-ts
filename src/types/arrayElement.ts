import { ElementStates } from "./element-states";
export type TArrayElement = { value: number, state: ElementStates };
