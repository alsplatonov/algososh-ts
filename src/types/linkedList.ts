import { ElementStates } from "./element-states";

export interface IListState<T> {
  item: T;
  state: ElementStates;
  insertProgress: boolean;
  removeProgress: boolean;
  tempItem: T;
}