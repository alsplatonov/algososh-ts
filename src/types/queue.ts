import { ElementStates } from "./element-states";

// Определяем интерфейс для элемента очереди
export interface IQueueItem<T> {
  data: T | null;
  state: ElementStates;
}