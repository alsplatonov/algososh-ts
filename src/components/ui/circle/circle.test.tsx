import React from 'react';
import { render } from '@testing-library/react';
import { Circle } from './circle';
import { ElementStates } from '../../../types/element-states';

describe('Circle component snapshots', () => {
  it('renders without a letter', () => {
    const { asFragment } = render(<Circle />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with letter', () => {
    const { asFragment } = render(<Circle letter="A" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with head as string', () => {
    const { asFragment } = render(<Circle head="Head" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with head as React element', () => {
    const { asFragment } = render(<Circle head={<div>Head Element</div>} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with tail as string', () => {
    const { asFragment } = render(<Circle tail="Tail" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with tail as React element', () => {
    const { asFragment } = render(<Circle tail={<div>Tail Element</div>} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with index', () => {
    const { asFragment } = render(<Circle index={5} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders as small', () => {
    const { asFragment } = render(<Circle isSmall={true} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders in default state', () => {
    const { asFragment } = render(<Circle state={ElementStates.Default} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders in changing state', () => {
    const { asFragment } = render(<Circle state={ElementStates.Changing} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders in modified state', () => {
    const { asFragment } = render(<Circle state={ElementStates.Modified} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
