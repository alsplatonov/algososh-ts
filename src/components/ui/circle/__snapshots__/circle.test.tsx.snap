// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component snapshots renders as small 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle  small default"
      data-testid="circleUpSmall"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshots renders in changing state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   changing"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshots renders in default state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshots renders in modified state 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   modified"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshots renders with head as React element 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    >
      <div>
        Head Element
      </div>
    </div>
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshots renders with head as string 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
      data-testid="circle-head"
    >
      Head
    </div>
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshots renders with index 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      5
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshots renders with letter 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        A
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshots renders with tail as React element 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    >
      <div>
        Tail Element
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshots renders with tail as string 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
      data-testid="circle-tail"
    >
      Tail
    </div>
  </div>
</DocumentFragment>
`;

exports[`Circle component snapshots renders without a letter 1`] = `
<DocumentFragment>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</DocumentFragment>
`;
